algorithms task (6)/student/include/impl.h

```diff
@@ -19,4 +21,16 @@
  */
 
 template <typename Sequence, typename FwIt>
-void insert_to_sorted(Sequence& target, const FwIt sourceBegin, const FwIt sourceEnd);
+void insert_to_sorted(Sequence& target, const FwIt sourceBegin, const FwIt sourceEnd)
+{
+    static_assert(std::is_const<typename std::remove_reference<
+                      typename std::iterator_traits<FwIt>::reference>::type>::value,
+                  "FwIt must point to const data");
+
+    // Use std::upper_bound() for stable order of source values.
+
+    std::for_each(sourceBegin, sourceEnd, [&target](typename std::iterator_traits<FwIt>::reference sourceValue)
```

Comment:
please avoid for_each

Reply:
but here I think it is acceptable

Reply:
I believe this is the only reason why std::for_each() is still alive, because for (auto it = begin; it != end; ++it) doesn't look better at all than std::for_each(begin, end, lambda);

Reply:
you always should prefer range for, but here you have only iterators, so it is ok